controllers/slot_controller.go

```diff
@@ -0,0 +1,50 @@
+package controllers
+
+import (
+	"airport-app-backend/repositories"
+	"net/http"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+)
+
+var ALL_SLOTS = "ALL"
+
+type SlotController struct {
+	repository repositories.ISlotRepository
+}
+
+func NewSlotController(repository repositories.ISlotRepository) *SlotController {
+	return &SlotController{
+		repository: repository,
+	}
+}
+
+// @Summary Get all Slots
+// @Router /slots [get]
+// @Description get all the slots
+// @ID get-all-slots
+// @Tags slot
+// @Produce  json
+// @Param   page        query    int     false        "Page number (default = 1)"
+// @Param   status        query    string     false        "Status of the slot. [Reserved, Booked, Available] (default = ALL)"
+// @Success 200  "ok"
+// @Failure 500 "Internal server error"
+func (sc SlotController) HandleGetAllSlots(ctx *gin.Context) {
+	page, _ := strconv.Atoi(ctx.Query("page"))
+	if page < 0 {
+		ctx.JSON(400, gin.H{"msg": "Page number must be greater than 0"})
+		return
+	}
+	status := ctx.Query("status")
+	if len(status) <= 0 {
+		status = ALL_SLOTS
```

Comment:
dont go with a string - which can be anything. For our purposes, we only want to filter `isAvailable` true/false (boolean ie only 2 possible values)

Reply:
Also, since you have implemented a filter, where are the appropriate tests?

Reply:
also, availability check/filter can be combined with a time-period as well (just a suggestion for future enhancement)

Reply:
filter implementation tests come at the service layer or in our case repo layer

Reply:
you will be calling the service/repo with different arguments based on logic in the controller. so, that specific logic has to be tested only in the controller layer ie whether valueA vs valueB gets passed to service/repo. Isn't that right?